searchlib/src/vespa/searchlib/features/queryterm.cpp

```diff
@@ -3,45 +3,66 @@
 #include "queryterm.h"
 #include "utils.h"
 
-using namespace search::fef;
 using search::feature_t;
+using namespace search::fef;
 
 namespace search::features {
 
-QueryTerm::QueryTerm() :
-    _termData(nullptr),
-    _handle(IllegalHandle),
-    _significance(0),
-    _connectedness(0)
+QueryTerm
+QueryTermFactory::create(const IQueryEnvironment & env, uint32_t termIdx, bool lookupConnectedness)
 {
+    const ITermData *termData = env.getTerm(termIdx);
+    feature_t fallback = util::getSignificance(*termData);
+    feature_t significance = features::util::lookupSignificance(env, termIdx, fallback);
+    feature_t connectedness = 0;
+    if (lookupConnectedness) {
+        connectedness = util::lookupConnectedness(env, termIdx);
+    }
+    return QueryTerm(termData, significance, connectedness);
 }
 
-QueryTerm::QueryTerm(const ITermData * td, feature_t sig, feature_t con) :
-    _termData(td),
-    _handle(IllegalHandle),
-    _significance(sig),
-    _connectedness(con)
+QueryTermHelper::QueryTermHelper(const IQueryEnvironment &env)
+        : _fallBack(),
+          _queryTerms(lookupQueryTerms(env))
 {
+    if (_queryTerms == nullptr) {
+        _fallBack = createQueryTermvector(env);
+        _queryTerms = & _fallBack;
+    }
 }
 
-QueryTerm
-QueryTermFactory::create(const IQueryEnvironment & env,
-                         uint32_t termIdx,
-                         bool lookupSignificance,
-                         bool lookupConnectedness)
+namespace {
+
+using QueryTermVectorWrapper = AnyWrapper<QueryTermVector>;
+const vespalib::string QUERY_TERMS_KEY("queryvectorhelper.queryterms");
```

Comment:
'queryvectorhelper' is a bit confusing, consider using 'querytermhelper.queryterms' instead

Reply:
fixed